### PR TITLE
feat: NAT- Stateless, Stateful

### PR DESCRIPTION
**Pull request by royalpinto007**
Documentation for NAT- Stateless and Stateful.

### CODE REVIEW

**Review by taraschornyiplv — DISMISSED** [review]

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+layout: default
+---
+
+# Stateful Network Address Translation (NAT)
+
+Stateful Network Address Translation (NAT), also known as dynamic NAT, facilitates the mapping of multiple private IP addresses to public addresses.
+
+- It was introduced in the [Beeblebrox Release (v2)](https://github.com/dentproject/dentOS/releases/tag/v2.0) of DENT.
+
+Configuration of stateful NAT involves employing the `tc iproute2` tool alongside the `tc-ct` action. For dynamic NAT entry offloading, integrate the `tc-ct` action with the flower filter rule and ingress qdisc.
```
Comment (taskin0003) [inline NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md:15]: I do not think these are right. Or at least I do not see these actions used in the doc sections below. Could you please double check?

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+parent: Network Addressing and Filtering
+nav_order: 4
+layout: default
+---
+
+# Stateless Network Address Translation (NAT)
+
+Stateless Network Address Translation (NAT), also known as static NAT, establishes a permanent mapping between a private IP address and a single public address.
+
+To configure stateless NAT, the `tc iproute2` tool is utilized, along with the `tc-nat` action. This action is combined with the `flower` filter rule and `ingress` qdisc to enable static NAT entry offloading. Below is the command format:
```
Comment (taskin0003) [inline NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md:13]: I do not think these are right. Or at least I do not see these actions used in the doc sections below. Could you please double check?

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+### Example Configuration:
```
Comment (taskin0003) [inline NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md:17]: I see this example configuration, but I do not know if you've tested it. If you did, please add a section on how you verified this config.

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+Where:
+
+- `<OLD>` represents the IP address to be translated.
+- `<NEW>` represents the IP address to translate to.
+- `ingress` translates destination addresses (performs DNAT).
+- `egress` translates source addresses (performs SNAT).
+
+To implement NAT one-to-one mapping, two rules are required: one for private-to-public direction and another for public-to-private direction.
+
+### Example Configuration
```
Comment (taskin0003) [inline NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md:28]: I see this example configuration, but I do not know if you've tested it. If you did, please add a section on how you verified this config.

**Review by taskin0003 — CHANGES_REQUESTED** [review]

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md (hunk 1 above)
Comment (royalpinto007) [reply NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md:15]: I'm not sure about the release here, but this is something mentioned in the DENT roadmap. I can change that if you provide some info or simply remove this line.

Regarding the usage of commands, the use of `tc` (iproute2) and `tc-ct` has been documented. Example commands used:

```
tc filter add dev swp20 ingress proto ip pref 2 flower ct_state -trk action ct
tc filter add dev swp21 egress prio 10 proto ip flower action ct commit nat src addr 91.245.77.1 pipe action pass
```

Comment (taskin0003) [reply NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md:15]: Let's not delete these just simply verify this config works. Please add a graph of sample topology (you can use VMs) and verify the config works before deleting/modifying anything.

File: NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md (hunk 2 above)
Comment (taskin0003) [reply NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateless-NAT.md:13]: Nevermind, I do see action nat. Please add a graph of sample topology (you can use VMs) and verify the config works before deleting/modifying anything.

**Review by taskin0003 — APPROVED** [review]

**Review by taraschornyiplv — APPROVED** [review]

**Pull request merged**